coldfront/api/allocation/views.py

```diff
@@ -112,3 +140,49 @@ def get_queryset(self):
             allocation_user_attributes = allocation_user_attributes.filter(
                 allocation_user=allocation_user_pk)
         return allocation_user_attributes.order_by("-history_date")
+
+
+class ClusterAccessRequestViewSet(mixins.ListModelMixin,
+                                  mixins.RetrieveModelMixin,
+                                  mixins.UpdateModelMixin,
+                                  viewsets.GenericViewSet):
+    """A ViewSet for the ClusterAccessRequestViewSet model."""
+
+    filterset_class = ClusterAccessRequestFilter
+    http_method_names = ['get', 'patch']
+    permission_classes = [IsSuperuserOrStaff]
+    serializer_class = ClusterAccessRequestSerializer
+
+    def get_queryset(self):
+        return ClusterAccessRequest.objects.order_by('id')
+
+    def perform_update(self, serializer):
+        try:
+            with transaction.atomic():
+                # Pop read only fields
+                cluster_uid = serializer.validated_data.pop('cluster_uid', None)
+                username = serializer.validated_data.pop('username', None)
+                serializer.validated_data.pop('allocation_user', None)
+
+                instance = serializer.save()
+
+                if instance.status.name == 'Active':
+                    runner = ClusterAccessRequestCompleteRunner(instance)
+                    runner.run(cluster_uid=cluster_uid,
```

Comment:
I imagine this works as is, but the parameters are not named, and the arguments are in the wrong order.